backend/src/services/journal/journal.repo.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+class JournalRepo {
+  constructor(db) {
+    this.db = db;
+  }
+
+  /**
+   * Create a new journal entry for a user.
+   * @param {number} userId
+   * @param {number} courseId
+   * @param {string} title
+   * @param {string} content
+   * @returns {Promise<object>} created journal entry
+   */
+  async createJournalEntry(userId, courseId, title, content) {
+    // check if content is smaller than 1GB
+    if (Buffer.byteLength(content, 'utf8') > 1e9) {
+      throw new Error('Content size exceeds 1GB limit');
+    }
+    const entry = await this.db.journals.create({
+      data: {
+        student_id: userId,
+        course_id: courseId,
+        title: title,
+        content: content,
+      },
+    });
+    return entry;
+  }
+
+  /**
+   * get all journal entries for a specific course.
+   * @param {number} courseId
+   * @returns {Promise<Array>} list of journal entries
+   */
+  async getJournalsByCourseId(courseId) {
+    const entries = await this.db.journals.findMany({
+      where: {
+        course_id: courseId,
+        deleted_at: null,
+      },
+    });
+    return entries;
+  }
+
+  /**
+   * Get a journal entry by its ID.
+   * @param {number} journalId
+   * @returns {Promise<object>} journal entry
+   */
+  async getJournalById(journalId) {
+    const entry = await this.db.journals.findFirst({
+      where: {
+        id: journalId,
+        deleted_at: null,
+      },
+    });
+    return entry;
+  }
+
+  /**
+   * get journals by course id + user id
+   * @param {number} userId
+   * @param {number} courseId
+   * @returns {Promise<Array>} list of journal entries
+   */
+  async getJournalsByUserIdAndCourseId(userId, courseId) {
+    const entries = await this.db.journals.findMany({
+      where: {
+        student_id: userId,
+        course_id: courseId,
+        deleted_at: null,
+      },
+    });
+    return entries;
+  }
+
+  /**
+   * Update the latest journal entry for a user.
+   * @param {number} journal_id
+   * @param {string} title
+   * @param {string} content
+   * @returns {Promise<object>} updated journal entry
+   */
+  async updateJournalEntry(journal_id, title, content) {
+    const entry = await this.db.journals.updateMany({
```

Comment:
[Suggestion]
Use `update` instead of `updateMany` in order to get the updated record as the returned result of the query.
Similar idea for the `deleteJournalEntry` function.